docs/DeveloperGuide.adoc

```diff
@@ -315,33 +324,333 @@ _{More to be added}_
+
+*Main Success Scenario*
+
+1. User requests to remove a task from the dashboard
+2. User specifies the task ID
+3. +work updates the dashboard
++
+Use case ends
+
+*Extensions*
+
+2. User specifies an invalid task ID
+3.  +work shows an error message
++
+Use case ends
+
+[discrete]
+=== Use case:   Assign a deadline to a task
+
+*Main Success Scenario*
+
+1. User requests to assign a deadline to a task
+2. User specifies the task ID and the corresponding deadline
+3. +work updates the dashboard
++
+Use case ends
+
+*Extensions*
+
+2. User specifies an invalid task ID
+3. User inputs the deadline in an incorrect format
+3.  +work shows an error message
++
+Use case ends
+
+[discrete]
+=== Use case: Generate availability timings of team members
+
+*Main Success Scenario*
+
+1. User adds timetable of team mates
+2. User requests to generate availability timings of team members
+3. +work displays list of timings where the most number of team members are available
++
+Use case ends
+
+*Extensions*
+
+2. User specifies the time in an invalid format
+3. +work shows an error message
++
+Use case ends
+
+[discrete]
+=== Use case: Add a meeting to the timetable
+
+*Main Success Scenario*
+
+1. User requests to add a meeting
+2. User specifies a description and the time of the meeting
+3. +work updates the dashboard
++
+Use case ends
+
+*Extensions*
+
+2. User specifies the time in an invalid format
+3. +work shows an error message
++
+Use case ends
+
+[discrete]
+=== Use case: Add an item to the inventory
+
+*Main Success Scenario*
+
+1. User requests to add an item to the inventory
+2. User specifies the name and price of the inventory item as well as the ID of the member and task associated with the item
+3. +work updates the inventory
++
+Use case ends
+
+*Extensions*
+
+2. User specifies an invalid task or member ID
+3. +work shows an error message
++
+Use case ends
+
+
+[discrete]
+=== Use case: Delete an item from the inventory
+
+*Main Success Scenario*
+
+1. User requests to delete an item from the inventory
+2. User specifies the item ID
+3. +work updates the inventory
++
+Use case ends
+
+*Extensions*
+
+2. User specifies an invalid inventory ID
+3. +work shows an error message
++
+Use case ends
+
+[discrete]
+=== Use case: Generate an inventory report
+
+*Main Success Scenario*
+
+1. User requests to generate an inventory report
+2. User specifies whether inventory is generated based on the member or task
+3. +work displays the inventory report
++
+Use case ends
+
+
+[discrete]
+=== Use case: Delete person
+
+*Main Success Scenario*
+
+1. User requests to list persons
+2. +work shows a list of persons
+3. User requests to delete a specific person in the list
+4. +work deletes the person
++
+Use case ends
+
+*Extensions*
+
+2. The list is empty.
+Use case ends
+
+3. The given index is invalid.
+4. +work shoes an error message
++
+Use case resumes at step 2
 
 [appendix]
 == Non Functional Requirements
```

Comment:
1. Are the NFRs are really Non-Functional?
- Yes because it is not related to any function. They are function requirements related to performance, user-friendliness. 

2. Is each NFR well-defined (i.e., possible to decide when it has been met)?

- NFR 1 is a well-defined NFR, but very generic
- NFR 2: it is ambiguous as to what ‘noticeable sluggishness’ means. How many seconds/milliseconds of lag would it be before we notice the sluggishness?
- NFR 3: it is a measurable NFR. Since the wording is ‘accomplish most of the tasks faster using commands’, are there any specific tasks where it would take longer to accomplish?
- Other feedback: the NFRs are quite generic, and not specific to the application. 

3. Is each NFR reasonably achievable?
- NFRs are reasonably achievable

4. Are more relevant NFRs have been left out?
- Left out NFRs related to security. A possible NFR would be: only team leader can change the allocation of tasks to team members.